File: playground/src/main/java/com/jiamian/translation/entity/dto/ModelApiDTO.java
package com.jiamian.translation.entity.dto;

import com.google.common.collect.Lists;
import lombok.Data;

import java.util.List;

/**
 * @author DingGuangHui
 * @date 2023/3/28
 */
@Data
public class ModelApiDTO {
	private long id;
	private String name = "";
	private String description= "";
	private String type= "";
	private boolean poi = true;
	private boolean nsfw = false;
	private boolean allowCommercialUse = false;
	private boolean allowNoCredit = false;
	private boolean allowDerivatives = true;
	private boolean allowDifferentLicense = false;
	private StatusApiDTO stats = new StatusApiDTO();
	private CreatorApiDTO creator = new CreatorApiDTO();
	private List<String> tags = Lists.newArrayList();
	private List<ModelVersionsApiDTO> modelVersions = Lists.newArrayList();

}

File: playground/src/main/java/com/jiamian/translation/entity/dto/api/ModelApiDTO.kt
package com.jiamian.translation.entity.dto.api

import com.google.common.collect.Lists

/**
 * @author DingGuangHui
 * @date 2023/3/28
 */

data class ModelApiDTO(var id: Long) {

     var name = ""
     var description = ""
     var type = ""
     var poi = true
     var nsfw = false
     var allowCommercialUse = false
     var allowNoCredit = false
     var allowDerivatives = true
     var allowDifferentLicense = false
     var stats = StatusApiDTO()
     var creator = CreatorApiDTO()
     var tags: List<String> = Lists.newArrayList()
     var modelVersions: List<ModelVersionsApiDTO> = Lists.newArrayList()
}